packages/analytics/amplify_analytics_pinpoint/lib/src/flutter_analytics_client.dart

```diff
@@ -0,0 +1,41 @@
+import 'package:amplify_analytics_pinpoint/src/device_context_info_provider/flutter_device_context_info_provider.dart';
+import 'package:amplify_analytics_pinpoint/src/flutter_endpoint_info_store_manager.dart';
+import 'package:amplify_analytics_pinpoint_dart/amplify_analytics_pinpoint_dart.dart';
+
+/// {@template amplify_analytics_pinpoint_dart.flutter_endpoint_store_manager}
+/// Manages retrieval, storage, and management of Pinpoint Endpoint id and fields
+/// Auto injects Flutter DI
+/// {@endtemplate}
+class FlutterAnalyticsClient extends AnalyticsClient {
```

Comment:
This client class will be used by Analytics, Push Notifications and any further category that requires Endpoint/Events with Pinpoint. 

It is designed to encapsulate as much internal initialization logic as possible to minimize exports. 

As a result we no longer need to export: 
'FlutterDeviceContextInfoProvider' 
'AmplifySecureStorage' 

For users to be able to instance 'AnalyticsClient'.